homeassistant/components/asuswrt/bridge.py

```diff
@@ -0,0 +1,299 @@
+
+    @abstractmethod
+    async def async_get_available_sensors(self) -> dict[str, dict[str, Any]]:
+        """Return a dictionary of available sensors for this bridge."""
+
+
+class AsusWrtLegacyBridge(AsusWrtBridge):
+    """The Bridge that use legacy library."""
+
+    def __init__(
+        self, conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> None:
+        """Initialize Bridge."""
+        super().__init__(conf[CONF_HOST])
+        self._protocol: str = conf[CONF_PROTOCOL]
+        self._api: AsusWrtLegacy = self._get_api(conf, options)
+
+    @staticmethod
+    def _get_api(
+        conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> AsusWrtLegacy:
+        """Get the AsusWrtLegacy API."""
+        opt = options or {}
+
+        return AsusWrtLegacy(
+            conf[CONF_HOST],
+            conf.get(CONF_PORT),
+            conf[CONF_PROTOCOL] == PROTOCOL_TELNET,
+            conf[CONF_USERNAME],
+            conf.get(CONF_PASSWORD, ""),
+            conf.get(CONF_SSH_KEY, ""),
+            conf[CONF_MODE],
+            opt.get(CONF_REQUIRE_IP, True),
+            interface=opt.get(CONF_INTERFACE, DEFAULT_INTERFACE),
+            dnsmasq=opt.get(CONF_DNSMASQ, DEFAULT_DNSMASQ),
+        )
+
+    @property
+    def is_connected(self) -> bool:
+        """Get connected status."""
+        return bool(self._api.is_connected)
+
+    async def async_connect(self) -> None:
+        """Connect to the device."""
+        try:
+            await self._api.connection.async_connect()
+        except OSError as exc:
+            raise ConfigEntryNotReady from exc
+
+        # get main router properties
+        await self._get_label_mac()
+        await self._get_firmware()
+        await self._get_model()
+
+    async def async_disconnect(self) -> None:
+        """Disconnect to the device."""
+        if self._api is not None and self._protocol == PROTOCOL_TELNET:
+            self._api.connection.disconnect()
+
+    async def async_get_connected_devices(self) -> dict[str, WrtDevice]:
+        """Get list of connected devices."""
+        try:
+            api_devices = await self._api.async_get_connected_devices()
+        except OSError as exc:
+            raise UpdateFailed(exc) from exc
+        return {
+            format_mac(mac): WrtDevice(dev.ip, dev.name, None)
+            for mac, dev in api_devices.items()
+        }
+
+    async def _get_nvram_info(self, info_type: str) -> dict[str, Any]:
+        """Get AsusWrt router info from nvram."""
+        info = {}
+        try:
+            info = await self._api.async_get_nvram(info_type)
+        except OSError as exc:
+            _LOGGER.warning(
+                "Error calling method async_get_nvram(%s): %s", info_type, exc
+            )
+
+        return info
+
+    async def _get_label_mac(self) -> None:
+        """Get label mac information."""
+        if self._label_mac is None:
+            self._label_mac = ""
```

Comment:
This is to avoid that the functions try to get the information if the information are not available after first try. If we leave the attributes = None, every time the router reconnect will try again to get this information from NVRam. So is to avoid useless requests, but I can also remove if you prefer.

Reply:
The point here is that `self._label_mac` will be always `None`  if mac is not returned by `self._get_nvram_info("LABEL_MAC")`, and so function will be called again. But I understand that you don't like force the value to empty string, I think I can just remove that part and let function be re-called in case of re-connection.

Reply:
Maybe I'm missing something... do some devices always return `None` for `self._get_nvram_info("LABEL_MAC")`?

Reply:
Sorry, you are right. It will return `{}` if nvram info is not available. I'll change as suggested.

Reply:
Ok, not exactly, but I will make a change, that you will give me your comment

Reply:
I made the change, but this should end with method called again when we call connect. This probably do not happen very often, so I think at the end is not a problem and code is more clear.

Reply:
Because we still need this in `async_connect()`:
```python
        # get main router properties
        if self._label_mac is None:
            await self._get_label_mac()
        if self._firmware is None:        
            await self._get_firmware()
        if self._model is None:
            await self._get_model()
```

Reply:
Yes, understand your point, I will add this. But in case that the method return an empty `dict` because property is not available on the router nvram, the attribute value remain `None` and the related method is called again. For that reason I was assigning "" at first call.